### PR TITLE
Debian packaging

### LINKED ISSUE #2186: Debian packaging
Build Debian packages and release them through your own repos; easing the usage for end users.

### LINKED ISSUE #2188: Mycroft Build Process improvements
- Update all dependencies to the latest version, for more compatibility and stability across  systems.
- Make the dev_setup.sh script runnable as root by default as all packaged versions of mycroft have to run it as root.(as a configuration script)

### LINKED ISSUE #2191: Make start-mycroft.sh independent from dev_setup.sh
The dev_setup.sh script included in the source release would not function as it requires the directory to be a git repo.

It has to be fixed in order for the source release to be packaged.

### PR DESCRIPTION
**Pull request by lyndalopez544**
## Description
closes #2191, closes #2188, closes #2186

I had to remove a few garbage from start-mycroft.sh too.
the build_deb script builds the Debian packages in the upper directory.

Contributor license agreement signed?
CLA [ Yes ]

### CODE REVIEW

**Comment by devs-mycroft** [comment]
Hello, @aurora-of-earth, thank you for helping with the Mycroft project! We welcome everyone
into the community and greatly appreciate your help as we work to build an AI
for Everyone.

To protect yourself, the project, and users of Mycroft technologies we require
a Contributor Licensing Agreement (CLA) before accepting any code
contribution. This agreement makes it crystal clear that along with your
code you are offering a license to use it within the confines of this project.
You retain ownership of the code, this is just a license.

Please visit https://mycroft.ai/cla to initiate this one-time signing. Thank
you!

**Comment by forslund** [comment]
Thanks again,

As previously stated we will not merge this into the mycroft-core repo but once I've verified that we can open the packaging repo we could consider merging this there.

Did you check the dh-virtualenv based debian package I posted on the other PR? https://github.com/forslund/mycroft-desktop-deb

The modifications on start-mycroft.sh aren't really compatible with the github and picroft setups so we'd need to find another solution for that. Generally a packaged solution shouldn't use the start-mycroft.sh which is a convenience script for the developers.

File: debian/postinst [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ln -S $MYCROFT_BIN/mycroft-msm /bin/mycroft-msm
+ln -S $MYCROFT_BIN/mycroft-pip /bin/mycroft-pip
+ln -S $MYCROFT_BIN/mycroft-say-to /bin/mycroft-say-to
+ln -S $MYCROFT_BIN/mycroft-skill-testrunner /bin/mycroft-skill-testrunner
+ln -S $MYCROFT_BIN/mycroft-speak /bin/mycroft-speak
+ln -S $MYCROFT_BIN/mycroft-stop /bin/mycroft-stop
+ln -S $MYCROFT_BIN/mycroft-start /bin/mycroft-start
+
+cd /opt/mycroft/mycroft-core
+source venv-activate.sh
```
Comment (forslund) [inline debian/postinst]: You activate the venv but I don't see it being created?

File: debian/postinst [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+MYCROFT=/opt/mycroft
+MYCROFT_LOG=/var/log/mycroft
+MYCROFT_BIN=/opt/mycroft/mycroft-core/bin
+
+mkdir -p $M/skills $MYCROFT_LOG /etc/mycroft
```
Comment (forslund) [inline debian/postinst]: Should it be $MYCROFT/skills ?

File: debian/postinst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+chown -R root:root $MYCROFT
```
Comment (forslund) [inline debian/postinst]: How will skill installation work when run as a normal user?

**Comment by aurora-of-earth** [comment]
> Thanks again,
> 
> As previously stated we will not merge this into the mycroft-core repo but once I've verified that we can open the packaging repo we could consider merging this there.
> 
> Did you check the dh-virtualenv based debian package I posted on the other PR? https://github.com/forslund/mycroft-desktop-deb
> 
> The modifications on start-mycroft.sh aren't really compatible with the github and picroft setups so we'd need to find another solution for that. Generally a packaged solution shouldn't use the start-mycroft.sh which is a convenience script for the developers.

What that even is?
My PR is a standard Debian packaging which does fully integrate with Debian.
That's how every Debian package is made.

**Comment by forslund** [comment]
dh-virtualenv is an extension to debhelper for handling python virtualenvironments.

File: debian/postinst (hunk 1 above)
Comment (aurora-of-earth) [reply debian/postinst]: This is the venv which is already provided in the mycroft-core repo.
It just cds into the installed mycroft-core repo and creates a venv right there.
And as I see every script in the bin folder do actually activate the venv before doing anything else.

Comment (forslund) [reply debian/postinst]: Ok so the virtualenv is prebuilt and included in the package then. gotcha.

File: debian/postinst (hunk 3 above)
Comment (aurora-of-earth) [reply debian/postinst]: users would install skills with root access but use the skills without root access.
The parent issue is that Mycroft doesn't work as root.

**Comment by aurora-of-earth** [comment]
@forslund 
> dh-virtualenv is an extension to debhelper for handling python virtualenvironments. [Linky](https://github.com/spotify/dh-virtualenv/)

What are you saying?
It has nothing to do with dh-virtualenv.
Part of the Python requirements are just transformed into available Debian package and are put into debian/control file.
The rest of the Python dependencies which not not available as Debian packages in the Debian repo are actually installed immediately after the mycroft debian package is installed.

**Comment by devs-mycroft** [comment]
Hello, @lyndalopez544, thank you for helping with the Mycroft project! We welcome everyone
into the community and greatly appreciate your help as we work to build an AI
for Everyone.

To protect yourself, the project, and users of Mycroft technologies we require
a Contributor Licensing Agreement (CLA) before accepting any code
contribution. This agreement makes it crystal clear that along with your
code you are offering a license to use it within the confines of this project.
You retain ownership of the code, this is just a license.

Please visit https://mycroft.ai/cla to initiate this one-time signing. Thank
you!

**Comment by krisgesling** [comment]
Hi there, as we don't have a signed CLA and there hasn't been activity on this PR in a long time I'm going to close it.

If you think this is a mistake please get in touch.

**Pull request closed without merging**